frontend/src/pages/projects/screens/spawner/storage/StorageClassSelect.tsx

```diff
@@ -66,7 +69,9 @@ const StorageClassSelect: React.FC<StorageClassSelectProps> = ({
           <SplitItem>{config?.displayName || sc.metadata.name}</SplitItem>
           <SplitItem isFilled />
           <SplitItem>
-            {config?.isDefault && (
+            {/* If multiple storage classes have `isDefault` set to true, 
+            prioritize the one returned by useDefaultStorageClass() as the default class */}
+            {config?.isDefault === true && config.displayName === defaultConfigName && (
```

Comment:
```suggestion
            {sc.metadata.name === defaultSc.metadata.name && (
```

You should just compare it like this. using the display name is not safe becuase they are not checked to be unique